torch/_inductor/compile_fx.py

```diff
@@ -307,14 +307,30 @@ def compile_fx_inner(
             if isinstance(t, torch.Tensor)
         )
 
-        if (
-            set(graph.device_types) == {"cuda"}
-            and not graph.mutated_inputs
-            and not has_incompatible_cudagraph_ops(gm)
-            and not complex_memory_overlap_inputs
-            and all(isinstance(t, torch.Tensor) for t in example_inputs)
-            and (len(graph.device_idxs) == 1 or not config.triton.cudagraph_trees)
-        ):
+        cudagraph_tests = [
+            (set(graph.device_types) == {"cuda"}, "non-cuda device in graph"),
+            (not graph.mutated_inputs, "mutated inputs"),
+            (not has_incompatible_cudagraph_ops(gm), "incompatible ops"),
+            (not complex_memory_overlap_inputs, "complex memory overlap"),
+            (
+                all(
+                    isinstance(t, (torch.Tensor, torch.SymInt)) for t in example_inputs
+                ),
+                "non-Tensor inputs",
+            ),
+            (
+                (len(graph.device_idxs) == 1 or not config.triton.cudagraph_trees),
+                "multiple device indices without cudagraph_trees",
+            ),
+        ]
+        cudagraph_fail_reasons = [s for b, s in cudagraph_tests if not b]
+
+        if not cudagraph_fail_reasons:
+            # Force specialize all inputs so that CUDA graphs will work
+            for t in example_inputs:
+                if isinstance(t, torch.SymInt):
```

Comment:
do we need to do this for SymFloat as well ?

Reply:
The contract is that SymFloat will never be an input per https://docs.google.com/document/d/1INSCdYu1PxXcr43HrD82OudeEuS-qxQe1yZmLg2wy6A/edit#heading=h.aytveihghlgn